docs/references/execution-errors.mdx

```diff
@@ -23,4 +24,190 @@ To fix this error, consider:
+
+To fix this error, consider increasing the canister's reserved cycles limit, or moving to a subnet with lower memory usage.
+
+
+### Insufficient cycles in message memory grow
+The canister doesn't have enough cycles to allocate the memory required to send a message.
+
+An example of this error is:
+```
+  Canister cannot grow message memory by 10240 bytes due to insufficient cycles.
+```
+
+Sending a message to another canister requires reserving space in the subnet's memory usage for the message and its response. This error indicates that a canister doesn't have enough cycles to pay for this memory usage without freezing.
+
+To fix this error, top up the canister with more cycles on decrease its [freezing threshold][setting-freezing-threshold].
+
+
+
+### Wasm memory limit exceeded
+The canister tried to grow its Wasm heap memory beyond the limit imposed by its Wasm memory limit setting.
+
+An example of this error is:
+```
+  Canister exceeded its current Wasm memory limit of 2147483648 bytes. The peak Wasm memory usage was 2147485000 bytes. If the canister reaches 4GiB, then it may stop functioning and may become unrecoverable. Please reach out to the canister owner to investigate the reason for the increased memory usage. It might be necessary to move data from the Wasm memory to the stable memory. If such high Wasm memory usage is expected and safe, then the developer can increase the Wasm memory limit in the canister settings.
+```
+
+Canisters may impose limits on the amount of Wasm heap memory they are allowed to use in order to prevent them from using the full 4 GiB. This is desirable because a canister that uses the full 4 GiB may to be able to upgrade if the pre-upgrade hook requires allocating more heap memory.
+
+
+To diagnose this error use the [`dfx canister status`][dfx-canister-status] command or the [`canister_status` API][canister-status-api] to check the canister's Wasm memory limit. If the memory usage is unexpected, try using [canbench][canbench] to determine if there is a memory leak.
+
+To fix this error, consider re-architecting the canister to use stable memory instead of heap memory or sharding the data across multiple canisters. Raising the memory limit may be another option if you are certain the canister can use more memory and still be upgradeable.
+
+
+
+
+[limits]: https://internetcomputer.org/docs/current/developer-docs/smart-contracts/maintain/resource-limits#resource-constraints-and-limits
+
+[canbench]: https://github.com/dfinity/dfxvm?tab=readme-ov-file#installation
+
+[dfx-canister-status]: https://internetcomputer.org/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-status
+
+[canister-status-api]: https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-canister_status
+
+[setting-freezing-threshold]: https://internetcomputer.org/docs/current/tutorials/hackathon-prep-course/managing-canisters/#setting-the-canisters-freezing-threshold
```

Comment:
```suggestion

```

Reply:
Is there a reason we don't want to have the commonly used references at the end? I was thinking it's easier to just change them in one spot if they need to be updated later.

Reply:
This isn't typically the format/style we use across the docs, so for consistency I recommend we format the links in the same in-line style that the rest of the docs use, but you can leave it as-is if you prefer that.

Reply:
Well if it's up to me then I'd definitely keep it like this. I also like that it's just easier to read and write the raw md with fewer long urls everywhere.